cirq/sim/density_matrix_simulator.py

```diff
@@ -620,6 +620,11 @@ def _value_equality_values_(self):
                         sorted(self.measurements.items())}
         return (self.params, measurements, self._final_simulator_state)
 
+    def __str__(self):
+        samples = super().__str__()
+        return 'measurements: {}\nfinal density matrix:\n{}'.format(
+            samples, str(self.final_density_matrix))
```

Comment:
I'm not sure if calling `str` on `final_density_matrix` is necessary here with the call to `.format` and all. What do you think @cduck?

Reply:
`str` isn't needed because format does this for you when you specify `'{}'` or `'{!s}'`.  Use `'{!r}'` if you want the repr.

Reply:
Cool, done.